crates/nu-parser/src/parser.rs

```diff
@@ -5251,6 +5256,36 @@ pub fn parse_builtin_commands(
     }
 }
 
+fn balanced_pair_between_qoutes(open: u8, closed: u8, input: &[u8]) -> bool {
+    let mut stack = Vec::new();
+    let mut escaped = false;
+    let mut in_quotes = false;
+
+    for c in input.iter() {
+        if escaped {
+            escaped = false;
+        } else if *c == b'\\' {
+            escaped = true;
+        } else if *c == b'"' {
+            in_quotes = !in_quotes;
+        } else if !in_quotes {
+            if *c == open {
+                stack.push(*c);
+            } else if *c == closed {
+                if let Some(last) = stack.pop() {
+                    if last != open {
```

Comment:
When will `last != open`?  It seems that stack only contains `open`

Reply:
Sorry, I still don't get the answer...

Reply:
Weird, my reply was missed somehow... Let me explain this one more time:

yes it is needed because this is a generic function, if the input `open` and `closed` do not match, this function might return a wrong result, consider [demo](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6466303ba4d13e98d5cfd4f725155d83)

Reply:
Thanks for your demo code, but when I running it, it returns `true` rather than `false`.

For your example: stack contains `{` before pop, and when we meet `]`, we pop from stack, then the last one is `{`, which equals to `open`...

Reply:
Did I blindly read `true` to `false` in my mind :(